python/cudf/cudf/core/column/column.py

```diff
@@ -1006,6 +1014,8 @@ def astype(self, dtype, **kwargs):
                     "Casting list columns not currently supported"
                 )
             return self
+        elif is_interval_dtype(self.dtype):
+            return self
```

Comment:
As with lists above, we should check that `self.dtype == dtype` and raise a `NotImplementedError` otherwise.